lua/RenderEvents.cpp

```diff
@@ -51,6 +51,17 @@ HOOK_METHOD(CommandGui, RenderPlayerShip, (Point &shipCenter, float jumpScale) -
     Global::GetInstance()->getLuaContext()->getLibScript()->call_on_render_event_post_callbacks(RenderEvents::LAYER_PLAYER, std::abs(idx), 0);
 }
 
+HOOK_METHOD_PRIORITY(Ship, OnRenderSparks, -1000, () -> void)
```

Comment:
Ran the script

Reply:
And again in the lua folder

Reply:
There's already an OnRenderSparks hook in CustomShips, the priority here is to ensure it's executed last.

Reply:
Is this defined *after* the other hook? If so, take out the priority as the order of definition already implies a priority.

Or if needs to be before just that one hook then it can be defined before it.

If it needs to be *globally* before or after all hooks for that method, that's when priority is used.

Reply:
Looking at the CMake file, it looks like this source file _is_ loaded after the one with the other hook. Fixed.